api/src/main/java/org/apache/cloudstack/api/BaseListDomainResourcesCmd.java

```diff
@@ -42,7 +45,10 @@ public boolean listAll() {
     @Override
     public boolean isRecursive() {
         if (listAll()) {
-            return recursive == null ? true : recursive;
+            Account caller = CallContext.current().getCallingAccount();
+            if (caller.getType() != Account.Type.NORMAL) {
+                return recursive == null ? true : recursive;
```

Comment:
Could be simplified to
```suggestion
                return recursive == null || recursive;
```

Reply:
more concise, less readable /m no me gusta